superset/connectors/sqla/utils.py

```diff
@@ -123,12 +123,19 @@ def get_virtual_table_metadata(dataset: SqlaTable) -> list[ResultSetColumnType]:
                 level=ErrorLevel.ERROR,
             )
         )
-    return get_columns_description(
-        dataset.database,
-        dataset.catalog,
-        dataset.schema,
-        statements[0],
-    )
+    try:
+        column_description = get_columns_description(
+            dataset.database,
+            dataset.catalog,
+            dataset.schema,
+            statements[0],
+        )
+        return column_description
+
+    except SupersetGenericDBErrorException as ex:
```

Comment:
Why is this necessary? It seems like we're re-raising the previous error with the same error type.